Source/BugsnagMetadata.m

```diff
@@ -45,13 +45,47 @@ - (id)initWithDictionary:(NSDictionary *)dict {
     if (self = [super init]) {
         // Ensure that the instantiating dictionary is mutable.
         // Saves checks later.
-        self.dictionary = [dict mutableCopy];
+        self.dictionary = [self sanitizeDictionary:dict];
         self.stateEventBlocks = [NSMutableArray new];
     }
     [self notifyObservers];
     return self;
 }
 
+/**
+ * Sanitizes the given dictionary to prevent [NSNull null] values from being added
+ * to the metadata when deserializing a payload.
+ *
+ * @param dictionary the input dictionary
+ * @return a sanitized dictionary
+ */
+- (NSMutableDictionary *)sanitizeDictionary:(NSDictionary *)dictionary {
+    NSMutableDictionary *input;
+    if (dictionary != nil && [dictionary isKindOfClass:[NSDictionary class]]) {
```

Comment:
I believe you need to invert the check to `![dictionary isKindOfClass:[NSMutableDictionary class]]` as `NSDictionary ` extends `NSDictionary` so would always be true.